### PR TITLE
crimson: use Seastar API v6

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
this change breaks seastore related tests
```
The following tests FAILED:
	203 - unittest_transaction_manager (Failed)
	204 - unittest_btree_lba_manager (Failed)
```

**Comment by tchaikov** [comment]
jenkins test make check

**Review by cyx1231st — APPROVED** [review]
LGTM

**Review by xxhdx1985126 — APPROVED** [review]

File: src/crimson/os/seastore/lba_manager/btree/lba_btree_node_impl.cc [hunk 1]
```diff
@@ -180,56 +180,42 @@ LBAInternalNode::mutate_internal_address_ret LBAInternalNode::mutate_internal_ad
...
+      if (addr == L_ADDR_NULL) {
+        ++i;
+        return {};
+      } else {
+        return addr;
+      }
+    },
+    // TODO: GCC enters a dead loop if crimson::do_until() is used
+    //       or erroratorized future is returned
+    crimson::ct_error::assert_all{ "fix me - APIv6" });
```
Comment (rzarzynski) [inline src/crimson/os/seastore/lba_manager/btree/lba_btree_node_impl.cc:217]: ACK. Thanks for the comment.

File: src/crimson/common/errorator.h [hunk 2]
```diff
@@ -505,7 +505,10 @@ struct errorator {
                                                 AllowedErrors>),
                     "provided Error Visitor is not exhaustive");
 
-      using value_func_result_t = std::invoke_result_t<ValueFuncT, ValuesT&&...>;
+      using value_func_result_t =
+        typename std::conditional_t<std::is_void_v<ValueT>,
+				    std::invoke_result<ValueFuncT>,
+				    std::invoke_result<ValueFuncT, ValueT>>::type;
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply src/crimson/common/errorator.h:511]: seems SFINAE fails to work here. i have
```
../src/crimson/common/errorator.h:508:13: error: no type named ‘type’ in ‘struct std::invoke_result<crimson::os::CyanStore::readv(crimson::os::FuturizedStore::CollectionRef, const ghobject_t&, interval_s\
et<long unsigned int>&, uint32_t)::<lambda(auto:72&)> [with auto:72 = ceph::buffer::v15_2_0::list]::<lambda(auto:73&)> [with auto:73 = std::pair<const long unsigned int, long unsigned int>]::<lambda(auto\
:74)> >’
  508 |       using value_func_result_t =
      |             ^~~~~~~~~~~~~~~~~~~
../src/crimson/common/errorator.h:514:13: error: no type named ‘type’ in ‘struct std::invoke_result<crimson::os::CyanStore::readv(crimson::os::FuturizedStore::CollectionRef, const ghobject_t&, interval_s\
et<long unsigned int>&, uint32_t)::<lambda(auto:72&)> [with auto:72 = ceph::buffer::v15_2_0::list]::<lambda(auto:73&)> [with auto:73 = std::pair<const long unsigned int, long unsigned int>]::<lambda(auto\
:74)> >’
  514 |       using value_func_errorator_t = get_errorator_t<value_func_result_t>;
      |             ^~~~~~~~~~~~~~~~~~~~~~
../src/crimson/common/errorator.h:519:13: error: no type named ‘type’ in ‘struct std::invoke_result<crimson::os::CyanStore::readv(crimson::os::FuturizedStore::CollectionRef, const ghobject_t&, interval_s\
et<long unsigned int>&, uint32_t)::<lambda(auto:72&)> [with auto:72 = ceph::buffer::v15_2_0::list]::<lambda(auto:73&)> [with auto:73 = std::pair<const long unsigned int, long unsigned int>]::<lambda(auto\
:74)> >’
  519 |       using return_errorator_t = make_errorator_t<
      |             ^~~~~~~~~~~~~~~~~~
```
i think the reason is that `::type` is always evaluated if `std::invoke_result_t` is used even if the other (unused) template fails to be substituted .

**Pull request merged**